napari/layers/points/_points_mouse_bindings.py

```diff
@@ -68,6 +69,16 @@ def select(layer, event):
         coordinates = layer.world_to_data(event.position)
         # If not holding modifying selection and points selected then drag them
         if not modify_selection and len(layer.selected_data) > 0:
+            # only emit just before moving
+            if not is_moving:
+                layer.events.data(
+                    value=layer.data,
+                    action=ActionType.CHANGING,
+                    data_indices=tuple(
+                        layer.selected_data,
+                    ),
+                    vertex_indices=((),),
+                )
```

Comment:
Part of the function that should be split on smaller parts in the future. This change looks correctly but is not easy to load whole scope

Reply:
Yeah in general the different layers would benefit from refactoring. Part of that is on the roadmap.